sound/soc/sof/topology.c

```diff
@@ -951,13 +958,29 @@ static int sof_parse_tokens(struct snd_soc_component *scomp,
 		array = (struct snd_soc_tplg_vendor_array *)((u8 *)array
 			+ asize);
 
+		/* update parse position in destination */
+		if (found && !(found % count))
+			offset += destsize;
+
 		/* return when all tokens have been found */
-		if (found == count)
+		if (found == count * sets)
 			return 0;
 	}
+
 	return 0;
 }
 
+static int sof_parse_tokens(struct snd_soc_component *scomp,
+			    void *object,
+			    const struct sof_topology_token *tokens,
+			    int count,
+			    struct snd_soc_tplg_vendor_array *array,
+			    int priv_size)
+{
+	return sof_parse_token_sets(scomp, object, tokens, count, array,
+				    priv_size, 1, 0);
```

Comment:
@juimonen need a comment here about the use of 1 and 0 saying that sof_parse_tokens() is only used when the topology does not contain multiple set of the same tokens. So the number of sets is 1 and the size is irrelevant.

Reply:
ok

Reply:
@juimonen this will break DMIC pdm token parsing, so lets just squash the 2nd and 3rd commits

Reply:
ok